src/Microsoft.ManagedZlib/src/Deflater.cs

```diff
@@ -0,0 +1,273 @@
+                    break;
+
+                case CompressionLevel.NoCompression:
+                    zlibCompressionLevel = ManagedZLib.CompressionLevel.NoCompression;
+                    memLevel = ManagedZLib.Deflate_NoCompressionMemLevel;
+                    break;
+
+                case CompressionLevel.SmallestSize:
+                    zlibCompressionLevel = ManagedZLib.CompressionLevel.BestCompression;
+                    memLevel = ManagedZLib.Deflate_DefaultMemLevel;
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(compressionLevel));
+            }
+
+            ManagedZLib.CompressionStrategy strategy = ManagedZLib.CompressionStrategy.DefaultStrategy;
+
+            ZErrorCode errC;
+            try
+            {
+                errC = ManagedZLib.CreateZLibStreamForDeflate(out _zlibStream, zlibCompressionLevel,
+                                                             windowBits, memLevel, strategy);
+            }
+            catch (Exception cause)
+            {
+                throw new ZLibException("ZLibErrorDLLLoadError - The underlying compression routine could not be loaded correctly.", cause);
+            }
+
+            switch (errC)
+            {
+                case ZErrorCode.Ok:
+                    return;
+
+                case ZErrorCode.MemError:
+                    throw new ZLibException("ZLibErrorNotEnoughMemory - The underlying compression routine could not reserve sufficient memory.",
+                        "deflateInit2_", (int)errC, _zlibStream.GetErrorMessage());
+
+                case ZErrorCode.VersionError:
+                    throw new ZLibException("ZLibErrorVersionMismatch - The version of the underlying compression routine does not match expected version.",
+                        "deflateInit2_", (int)errC, _zlibStream.GetErrorMessage());
+
+                case ZErrorCode.StreamError:
+                    throw new ZLibException("ZLibErrorIncorrectInitParameters - The underlying compression routine received incorrect initialization parameters.",
+                        "deflateInit2_", (int)errC, _zlibStream.GetErrorMessage());
+
+                default:
+                    throw new ZLibException("ZLibErrorUnexpected - The underlying compression routine returned an unexpected error code.", "deflateInit2_",
+                        (int)errC, _zlibStream.GetErrorMessage());
+            }
+        }
+
+        ~Deflater()
+        {
+            Dispose(false);
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (!_isDisposed)
+            {
+                if (disposing) {
+                    //_zlibStream.Dispose(); //Vivi's note: DISPOSE  OF ZSTREAM *dispose method not yet imlemented
```

Comment:
Are we still going to need to dispose ZlibStream? It seems you might be able to find a way to remove that class altogether, so probably this Dispose method could end up being empty.

Reply:
Yes, so far, ZlibStream seems not to be useful anymore.